app/src/main/java/com/example/next_show/data/ShowFilter.java

```diff
@@ -92,82 +82,96 @@ public void removeFromYear(String y) {
         filters.get(YEAR).remove(y);
     }
 
-    public void updateShowsList(List<Show> newShows) {
-        showsList.addAll(newShows);
-    }
-
-    public void setShowsList(List<Show> newShows) {
-        allShows = newShows;
+    public void addAll(List<Show> newShows) {
+        allShows.addAll(newShows);
     }
 
     public List<Show> getAllShows() {
         return allShows;
     }
 
+    public void clear() {
+        allShows.clear();
+    }
+
     public List<Show> filterShows(List<Show> shows) {
-        List<Show> filteredShows = shows;
+        // filter by genre first
+        Set<String> genreQueries = filters.get(GENRE);
+        List<Show> genreMatchedShows  = getGenreMatch(shows, genreQueries);
+
+        if(genreMatchedShows.isEmpty()) {
+            genreMatchedShows = shows;
+        }
 
-        // filter by network (if appplicable)
-        List<String> networkQueries = filters.get(NETWORK);
-        filteredShows = filterNetwork(filteredShows, networkQueries);
+        // queries for other attributes
+        Set<String> networkQueries = filters.get(NETWORK);
+        Set<String> yearQueries = filters.get(YEAR);
 
-        // filter by genre (if appplicable)
-        List<String> genreQueries = filters.get(GENRE);
-        filteredShows = filterGenre(filteredShows, genreQueries);
+        List<Show> filteredShows = new ArrayList<>();
 
-        // filter by year (if appplicable)
-        List<String> yearQueries = filters.get(YEAR);
-        filteredShows = filterYear(filteredShows, yearQueries);
+        for (Show s: genreMatchedShows) {
+            // check network and year
+            if (matchedNetworkFilter(s, networkQueries) && matchedYearFilter(s, yearQueries)) {
+                filteredShows.add(s);
+            }
+        }
```

Comment:
Much better!  One more thing you could do is also pull this out so that you do the genre match as a boolean on a per show basis in here as well.

You can do that by:
- Having the `getGenreMatch` utilize a method similar to `matchedNetworkQueries` which checks against a single show
- Then in here, instead of needing 2 iterations through the list, you can get down to 2 by just checking against a single show like you do in the highlighted for loop here (so using the new `matchedGenreQueries` or whatever you name that function